arch/x86/core/reboot_rst_cnt.c

```diff
@@ -20,21 +21,14 @@
 #define X86_RST_CNT_CPU_RST 0x4
 #define X86_RST_CNT_FULL_RST 0x08
 
-static inline void cold_reboot(void)
+void z_sys_reboot(enum sys_reboot_type type)
 {
-	uint8_t reset_value = X86_RST_CNT_CPU_RST | X86_RST_CNT_SYS_RST |
-				X86_RST_CNT_FULL_RST;
-	sys_out8(reset_value, X86_RST_CNT_REG);
-}
+	ARG_UNUSED(type);
```

Comment:
It is probably better to add an assert here instead of just change the implementation. It may help catch possible wrong usage.
`__ASSERT(type == SYS_REBOOT_COLD)`

Reply:
the API contract has been refined, please check https://github.com/zephyrproject-rtos/zephyr/pull/60920/commits/dfb5e2508664dcd7d7ae9c9ff9bae751d7768971

Reply:
Looked it, doesn't it make sense to at least log on targets that were doing distinct things before ?

Reply:
anyway, that is minor.